docs/dev/Kibana-Reporting-Design-Proposal.md

```diff
@@ -20,43 +21,111 @@ Kibana Reports for Open Distro allows ‘Report Owner’ (engineers, including b
-        * As a user, I should be able to archive reports. 
-            * Delete option should not be there as it might lead to complications such as one wherein the user who receives the reports deletes them and says that they never received the reports
-        * As a user, I should be able to see a list of reports archived for the past 100 days. NOTE: Reports will be stored as hyperlinks/urls and NOT as actual reports. If the data is available at the time when the url is clicked, report will be created and can be viewed in a browser and can be downloaded.
-
-### Functional 
-
+1. As a user, I want to see a 'Share >' 'As a report' on the share menu, and a 'Download > ' 'PDF, CSV, PNG' contextually button in the  Dashboard, Visualizations, and Discover Tabs.
+1. As a user, I should be able to export and share reports only if I have the permissions to do so. Else, an error should be thrown.
+1. As a user, when I click ‘Share > As a report’ I should be able to configure, save, export and share the report.
+1. As a user, when I click ‘Share > As a report’, the system should allow me to share the dashboard or saved search or visualizations that I am currently working on as a report.
+
+    Here is the matrix showing various panels and what kind of export option should be available.        
+    Kibana panel | PDF           | PNG           |Hyperlink (HTML) | CSV/Excel
+    ------------ | ------------- | ------------- | ------------- | -------------
+    Dashboards | Yes | Yes | Yes | No
+    Visualizations | Yes |  Yes |Yes | No
+    Discover (saved search) | No | No | Yes | Yes
+
+### In-context reports
+1. As a user, I can download a PDF, PNG, CSV, XLS file in-context from any available panel (e.g. Dashboards, visualizations, saved searches, etc).
+1. As a user, any time I download a report from any available panel (e.g. Dashboards, visualizations, saved searches, etc), a report will be auto-generated in the reports list.
+1. As a user, I can create a report definition from the share menu of any available panel (e.g. Dashboards, visualizations, saved searches, etc).
+
+### Reporting Landing Page
+#### Reports
+1. As a user, I should see Name, Type, Sender, Recepient(s), Source, Last updated, State, Download on the `Reports` list.
+1. As a user, I should be able to filter my reports list by sender, receiver, status, and type.
+1. As a user, I should be able to search my reports by report name.
+1. As a user, I should be able to view a report details page for any report on the list.
+1. As a user, I should see my reports ordered in reverse chronological order by default.
+1. As a user, I should be able to download the default file format from the reports list.
+1. As a user, I should be able to see a list of reports for the past 100 days.
+1. As a user, archived reports should not be visible by default (the default filter should exclude archived reports) (p2)
```

Comment:
Clarity on archive: archive is a status- shouldn't be able to see by default but a user should be able to filter by status to see archived reports

Reply:
correct